core/templates/dev/head/pages/admin/roles_tab/roles_tab_directive.html

```diff
@@ -0,0 +1,96 @@
+<style>
+    .form-horizontal .control-label {
+        text-align: left;
+    }
+</style>
+
+<div class="container-fluid">
+  <div class="row">
+    <br>
+    <h3 class="text-center text-danger">NOTE: Please don't modify roles from this page yet.</h3>
+  </div>
+  <div class="row">
+    <div class="col-md-8">
+      <md-card class="oppia-page-card oppia-long-text">
+        <form class="form-horizontal" ng-submit="SubmitRoleViewForm(view_form_values)">
+          <legend class="text-center">View User Roles</legend>
+          <div class="form-group">
+            <label class="col-md-4 col-lg-4 col-sm-4 control-label pull-left" for="view_form_values.method">
+              Select Method
+            </label>
+            <div class="col-md-8">
+              <select ng-model="view_form_values.method" class="form-control">
+                  <option value="role">By Role</option>
+                  <option value="username">By Username</option>
+              </select>
+            </div>
+          </div>
+
+          <div class="form-group" ng-if="view_form_values.method=='role'">
+            <label class="col-md-4 col-lg-4 col-sm-4 control-label pull-left">Select Role</label>
+            <div class="col-md-8 col-lg-8 col-sm-8">
+              <select ng-options="role_name as role_string for (role_name, role_string) in view_role_options" ng-model="view_form_values.role" class="form-control">
+              </select>
+            </div>
+          </div>
+
+          <div class="form-group" ng-if="view_form_values.method=='username'">
+            <label class="col-md-4 col-lg-4 col-sm-4 control-label">Enter Username</label>
+            <div class="col-md-8 col-lg-8 col-sm-8">
+              <input type="text" placeholder="Enter username" ng-model="view_form_values.username" class="form-control" />
+            </div>
+          </div>
+
+          <button type="submit" class="btn btn-success" ng-disabled="view_form_values.method==None || (view_form_values.role==undefined && view_form_values.username==undefined)" value="view role">View Role</button>
```

Comment:
A question: let's say the user chooses one view type and then types some value, then switches to the other view type. Is that value cleared (i.e. does switching back to the original view type result in a blank value field)? If not, is it at least cleared on submission?

I think I lean towards clearing the value only on submission, in which case you could do:
    ng-disabled="(viewFormValues.method === '...' && !viewFormValues.role) || (... && ...)"

I would incline against having viewFormValues.method ever be None (which, btw, isn't a thing in JS). Just pick a default value for it so that it's always valid.